backend/sftp/sftp.go

```diff
@@ -286,13 +287,33 @@ func (r *SFTP) Create() (backend.Blob, error) {
 	return &blob, nil
 }
 
+func Join(parts ...string) string {
+	if len(parts) == 0 {
+		return ""
+	}
+
+	result := strings.TrimRight(parts[0], "/")
+	if len(parts) == 1 {
+		return result
+	}
+
+	for _, s := range parts[1:] {
+		s = strings.TrimRight(s, "/")
+		if len(s) == 0 {
+			continue
+		}
+		result = result + "/" + s
+	}
+	return result
+}
+
 // Construct path for given backend.Type and name.
 func (r *SFTP) filename(t backend.Type, name string) string {
 	if t == backend.Config {
-		return filepath.Join(r.p, "config")
+		return Join(r.p + "config")
```

Comment:
Again, I think `+` may be wrong here, since `r.p` may not end with a slash. Call `Join(r.p, "config")` instead.